pkg/dao/dao.go

```diff
@@ -27,7 +23,7 @@ type Dao struct {
 	config    Config
```

Comment:
> I disagree slightly, the broker looks at the errors that are returned from the DAO as it could be etcd errors.

Fair. I don't think we should be doing that either since it's another abstraction leak for the same reasons :)

> I think it might be a good idea to define errors the broker knows about and keep the low level errors down in their own layers. i.e. broker shouldn't get an etcd error? but that would require some nasty handling since there are no Exceptions :)

I like this. Don't like go error handling, but we don't have a choice if this is the road we want to head down ¯\_(ツ)_/¯

Dao errors make sense to me. `NoConnection`, `FailedToRetrieve`, `NotFound`, things on that level.